streams/src/main/java/org/apache/kafka/streams/processor/internals/CompositeRestoreListener.java

```diff
@@ -45,41 +46,64 @@
         internalBatchingRestoreCallback = getBatchingRestoreCallback(stateRestoreCallback);
     }
 
+    /**
+     * @throws StreamsException if user provided {@link StateRestoreListener} raises an exception in
+     * {@link StateRestoreListener#onRestoreStart(TopicPartition, String, long, long)}
+     */
     @Override
     public void onRestoreStart(final TopicPartition topicPartition,
                                final String storeName,
                                final long startingOffset,
                                final long endingOffset) {
-        globalRestoreListener.onRestoreStart(topicPartition, storeName, startingOffset, endingOffset);
+        try {
+            userRestoreListener.onRestoreStart(topicPartition, storeName, startingOffset, endingOffset);
+        } catch (final Exception fatalUserException) {
```

Comment:
Could fatalUserException be StreamsException ?

Reply:
Only if the user code throws a `StreamsException` and this is not expected as nobody besides us should throw this (technically, we cannot disallow users to throw one, but it would not be correct on their side).